sql-plugin/src/main/spark340/scala/org/apache/spark/sql/hive/rapids/shims/GpuInsertIntoHiveTable.scala

```diff
@@ -0,0 +1,341 @@
+    rule: DataFromReplacementRule)
+  extends DataWritingCommandMeta[InsertIntoHiveTable](cmd, conf, parent, rule) {
+
+  private var fileFormat: Option[ColumnarFileFormat] = None
+
+  override def tagSelfForGpu(): Unit = {
+    // Only Hive delimited text writes are currently supported.
+    // Check whether that is the format currently in play.
+    fileFormat = GpuHiveTextFileFormat.tagGpuSupport(this)
+  }
+
+  override def convertToGpu(): GpuDataWritingCommand = {
+    GpuInsertIntoHiveTable(
+      table = wrapped.table,
+      partition = wrapped.partition,
+      fileFormat = this.fileFormat.get,
+      query = wrapped.query,
+      overwrite = wrapped.overwrite,
+      ifPartitionNotExists = wrapped.ifPartitionNotExists,
+      outputColumnNames = wrapped.outputColumnNames,
+      tmpLocation = cmd.hiveTmpPath.externalTempPath
+    )
+  }
+
+  def getCannotBeReplacedReasons(): Option[collection.mutable.Set[String]] = cannotBeReplacedReasons
+}
+
+case class GpuInsertIntoHiveTable(
+    table: CatalogTable,
+    partition: Map[String, Option[String]],
+    fileFormat: ColumnarFileFormat,
+    query: LogicalPlan,
+    overwrite: Boolean,
+    ifPartitionNotExists: Boolean,
+    outputColumnNames: Seq[String],
+    tmpLocation: Path) extends GpuSaveAsHiveFile {
+
+  /**
+   * Inserts all the rows in the table into Hive.  Row objects are properly serialized with the
+   * `org.apache.hadoop.hive.serde2.SerDe` and the
+   * `org.apache.hadoop.mapred.OutputFormat` provided by the table definition.
+   */
+  override def runColumnar(sparkSession: SparkSession, child: SparkPlan): Seq[ColumnarBatch] = {
+    val externalCatalog = sparkSession.sharedState.externalCatalog
+    val hadoopConf = sparkSession.sessionState.newHadoopConf()
+
+    val hiveQlTable = HiveClientImpl.toHiveTable(table)
+    // Have to pass the TableDesc object to RDD.mapPartitions and then instantiate new serializer
+    // instances within the closure, since Serializer is not serializable while TableDesc is.
+    val tableDesc = new TableDesc(
+      hiveQlTable.getInputFormatClass,
+      // The class of table should be org.apache.hadoop.hive.ql.metadata.Table because
+      // getOutputFormatClass will use HiveFileFormatUtils.getOutputFormatSubstitute to
+      // substitute some output formats, e.g. substituting SequenceFileOutputFormat to
+      // HiveSequenceFileOutputFormat.
+      hiveQlTable.getOutputFormatClass,
+      hiveQlTable.getMetadata
+    )
+
+    try {
+      processInsert(sparkSession,
+                    externalCatalog,
+                    hadoopConf,
+                    tableDesc,
+                    fileFormat,
+                    tmpLocation,
+                    child)
+    } finally {
+      // Attempt to delete the staging directory and the inclusive files. If failed, the files are
+      // expected to be dropped at the normal termination of VM since deleteOnExit is used.
+      HiveFileUtil.deleteExternalTmpPath(hadoopConf, tmpLocation)
```

Comment:
Change #3: Call a copy of the `deleteExternalTmpPath` code that existed prior to Spark 3.4.0